tuist/tuist
# Fix code signing issues in test targets

Resolves #3920

### Short description 📝

When using the `tuist test` command, we are explicitly disabling code signing by passing some extra arguments to the `xcodebuild` command.

One of the arguments passed is `CODE_SIGNING_ALLOWED=NO`. This is fine for most things, but if a test target has a test host that _does_ require code signing, passing this setting crashes the test host when it is run in the simulator.

You can see a more detailed explanation in #3920.

### How to test the changes locally 🧐

Here is a simple example project to test with: [CodeSignExample.zip](https://github.com/tuist/tuist/files/7790579/CodeSignExample.zip)

This project has some tests that require access to a device's Keychain, and as such, require a test host app to be running.

To test this, from the root of the project, run the following shell command:

```shell
xcodebuild \
      test \
      -project CodeSignExample.xcodeproj \
      -scheme CodeSignExample \
      -destination 'platform=iOS Simulator,name=iPhone 13,OS=15.2' \
      CODE_SIGN_IDENTITY= \
      CODE_SIGNING_REQUIRED=NO \
      CODE_SIGN_ENTITLEMENTS=
```

Note that the app is launched in the simulator, and the tests pass.

Next, run the same command, but add the `CODE_SIGNING_ALLOWED=NO` flag:

```shell
xcodebuild \
      test \
      -project CodeSignExample.xcodeproj \
      -scheme CodeSignExample \
      -destination 'platform=iOS Simulator,name=iPhone 13,OS=15.2' \
      CODE_SIGN_IDENTITY= \
      CODE_SIGNING_REQUIRED=NO \
      CODE_SIGN_ENTITLEMENTS= \
      CODE_SIGNING_ALLOWED=NO
```

Note that the app launches, but then crashes, and the tests fail.

I'm still not sure _exactly_ what's going on, but my hunch is like I said above: explicitly disallowing codesigning from the command line passes the behavior to the test host as well, thereby breaking any tests that have a test host that requires codesigning to function.

### Checklist ✅
